processor/src/decorators/adv_stack_injectors.rs

```diff
@@ -338,6 +338,147 @@ where
 
         Ok(())
     }
+
+    /// Pushes values onto the advice stack which are required for successful insertion of a
+    /// key-value pair into a Sparse Merkle Tree data structure.
+    ///
+    /// The Sparse Merkle Tree is tiered, meaning it will have leaf depths in `{16, 32, 48, 64}`.
+    ///
+    /// Inputs:
+    ///   Operand stack: [VALUE, KEY, ROOT, ...]
+    ///   Advice stack: [...]
+    ///
+    /// Outputs:
+    ///   Operand stack: [OLD_VALUE, NEW_ROOT, ...]
+    ///   Advice stack, depends on the type of insert:
+    ///   - Simple insert at depth 16: [d0, d1, ONE, ZERO]
+    ///   - Simple insert at depth 32 or 48: [d0, d1, ONE, ZERO, P_NODE, CHILD_1, CHILD_2]
+    ///   - Update of an existing leaf: [ONE, d0, d1, ONE, VALUE]
```

Comment:
Is the comment for complex insertion missing? Also, are the flag values consistent between the docs and the code?

Reply:
Yes, haven't added it yet as not sure how the advice stack will look like for this yet.